### PR TITLE
Property Resolution - part of fixing schema compilation

### PR DESCRIPTION
**Pull request by mbeckerle**
(This is a test. Code doesn't pass all tests yet - 34 failures. Trying out code-review mechanisms.)

Refactors property resolving onto elementrefs and grouprefs so
as to eliminate use of backrefs when accessing properties. (DOES NOT YET
ELIMINATE THE BACKREFS THO)

Removed alignment logic that was getting tripped up.

Converted OOLAGHost, SchemaComponent etc. down to LocalElementBase
into traits so that I can mix them the way I need to so that
Global elements aren't terms, etc.

### CODE REVIEW

File: daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/dpath/Expression.scala [hunk 1]
```diff
@@ -294,8 +294,8 @@ case class ComparisonExpression(op: String, adds: List[Expression])
       case ("=", _) => subsetError("Unsupported operation '%s'. Use 'eq' instead.", op)
       case ("!=", _) => subsetError("Unsupported operation '%s'. Use 'ne' instead.", op)
 
-      case("eq", HexBinary) => EQ_CompareByteArray
-      case("ne", HexBinary) => NE_CompareByteArray
+      case ("eq", HexBinary) => EQ_CompareByteArray
+      case ("ne", HexBinary) => NE_CompareByteArray
```
Comment (stevedlawrence) [inline daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/dpath/Expression.scala]: Testing commenting on a code hunk.

**Comment by stevedlawrence** [comment]
Testing general comment.

File: daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/dpath/Expression.scala (hunk 1 above)
Comment (mbeckerle) [reply daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/dpath/Expression.scala]: Thank you for that useful comment. (Just a test of comments and replies while I figure out this code-review system.)

Comment (VTGuy) [reply daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/dpath/Expression.scala]: Making a **bold** comment. :1st_place_medal:

File: daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/compiler/Compiler.scala [hunk 2]
```diff
@@ -67,6 +67,7 @@ import edu.illinois.ncsa.daffodil.processors.parsers.NotParsableParser
 import edu.illinois.ncsa.daffodil.processors.unparsers.NotUnparsableUnparser
```
Comment (mbeckerle) [inline daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/compiler/Compiler.scala]: Just using to try the "start a review" button, which is grey until you type a comment.

Comment (mbeckerle) [inline daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/compiler/Compiler.scala]: This is to figure out the difference between a review, and just comments on a patch set.

**Comment by mbeckerle** [comment]
Not merging. This was a test pull request.

**Pull request closed without merging**